packages/alfa-style/src/property/font.ts

```diff
@@ -0,0 +1,246 @@
+   * simplicity we assume a single base size for every generic font family.
+   */
+  const bases: { [P in Family.Generic["value"]]: number } = {
+    serif: 16,
+    "sans-serif": 16,
+    monospace: 13,
+    cursive: 16,
+    fantasy: 16
+  };
+
+  /**
+   * Scaling factors for absolute font sizes. In browsers, these factors are
+   * only used when the base font size are outside the range of 9 to 16 pixels.
+   * Within this range, a hardcoded lookup table is used for various legacy
+   * reasons but for simplicity we always rely on these scaling factors. The
+   * effect of this is that we will compute font sizes that are slightly larger
+   * than what browsers would render.
+   *
+   * @see https://drafts.csswg.org/css-fonts/#absolute-size-mapping
+   */
+  const factors: { [P in Size.Absolute["value"]]: number } = {
+    "xx-small": 3 / 5,
+    "x-small": 3 / 4,
+    small: 8 / 9,
+    medium: 1,
+    large: 6 / 5,
+    "x-large": 3 / 2,
+    "xx-large": 2 / 1,
+    "xxx-large": 3 / 1
+  };
+
+  /**
+   * @see https://drafts.csswg.org/css-fonts/#propdef-font-size
+   */
+  export const Size: Property<Size.Specified, Size.Computed> = Property.of(
+    Length.of(16, "px"),
+    either(
+      either(
+        Keyword.parse(
+          "xx-small",
+          "x-small",
+          "small",
+          "medium",
+          "large",
+          "x-large",
+          "xx-large",
+          "xxx-large"
+        ),
+        Keyword.parse("larger", "smaller")
+      ),
+      either(Percentage.parse, Length.parse)
+    ),
+    style =>
+      style.specified("font-size").map(size => {
+        switch (size.type) {
+          case "length":
+            return Resolver.length(size, style.parent);
+
+          case "percentage": {
+            const parent = style.parent.computed("font-size").value;
+
+            return Length.of(parent.value * size.value, parent.unit);
+          }
+
+          case "keyword": {
+            const parent = style.parent.computed("font-size").value;
+
+            switch (size.value) {
+              case "larger":
+                return Length.of(parent.value * 1.2, parent.unit);
+
+              case "smaller":
+                return Length.of(parent.value * 0.85, parent.unit);
+
+              default: {
+                const factor = factors[size.value];
+
+                const [family] = style.computed("font-family").value;
+
+                const base =
+                  family.type === "keyword" ? bases[family.value] : bases.serif;
+
+                return Length.of(factor * base, "px");
+              }
+            }
+          }
+        }
+      }),
+    {
+      inherits: true
+    }
+  );
+
+  export type Weight = Weight.Specified | Weight.Computed;
+
+  export namespace Weight {
+    export type Absolute = Keyword<"normal" | "bold">;
+
+    export type Relative = Keyword<"bolder" | "lighter">;
+
+    export type Specified = Absolute | Relative | Number;
+
+    export type Computed = Number;
+  }
```

Comment:
And I realise I got this wrong 😞 
The `Number` in `type Weight` was not here because it's `Weight.Computed` but because it's part of `Weight.Specified`… Which does make much more sense…

Anyway, I do think this is better at conveying the intend (it can be specified in some ways, and it will compute something, the fact that it can be specified in the exact same way it will be computed is now more explicitly serendipity 😀).